include/oneapi/dpl/pstl/parallel_backend_tbb.h

```diff
@@ -819,19 +819,24 @@ class __merge_func
     {
         template <typename Iterator>
         void
-        operator()(Iterator __first, Iterator __last)
+        operator()(Iterator __first, Iterator __last, ::std::false_type)
```

Comment:
What about code from algorithm_impl.h ?

            // The goal of this check - avoid to compile code with __brick_destroy call because
            // we have static_assert inside of __brick_destroy (if __cplusplus >= 201703L)
            if _ONEDPL_CONSTEXPR_IF (!::std::is_trivially_destructible<_T1>())
            {
                __par_backend::__parallel_for(::std::forward<_ExecutionPolicy>(__exec), __r + __n2, __r + __n1,
                                              [__is_vector](_T1* __i, _T1* __j)
                                              { __brick_destroy(__i, __j, __is_vector); });
            }